pkg/storage/mysql/mysql_test.go

```diff
@@ -25,6 +27,30 @@ func TestMySQLDatastore(t *testing.T) {
 	test.RunAllTests(t, ds)
 }
 
+func TestMigrate(t *testing.T) {
+	engine := "mysql"
+	// starts the container and runs migration up to the latest migration version available
+	testDatastore := storagefixtures.RunDatastoreTestContainer(t, engine)
+
+	uri := testDatastore.GetConnectionURI()
+	ds, err := New(uri, sqlcommon.NewConfig())
+	require.NoError(t, err)
+	defer ds.Close()
+
+	// going from version 3 to 4 when migration #4 doesn't exist is a no-op
+	version := testDatastore.GetDatabaseVersion() + 1
```

Comment:
Do we have control on the database version in the test?  (Reason I am asking is that how do we know that the current version is 4 and maximum version is 4 - which will result in a no-op).

Should we run downgrade until 0 (which you already do) and then run upgrade to maximum to ensure both upgrade and downgrade are tested?

Reply:
By default, when you create a test container, it runs all `up` migrations up to the latest one available.

So in the test i'm doing:
1. try to update to one version more -> no-op
2. run `down` until version 0 is reached